src/curator/harbor.py

```
@@ -0,0 +1,94 @@
+                status_code=404,
+                detail="No build metadata found in the latest artifact"
+            )
+
+        build_metadata = BuildPackMetadata.parse_raw(build_metadata_str)
+        return build_metadata.processes
+
+    except httpx.HTTPStatusError as e:
+        raise HTTPException(
+            status_code=e.response.status_code,
+            detail=f"Failed to fetch from Harbor: {str(e)}"
+        )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)